src/asmcnc/apps/maintenance_app/widget_maintenance_spindle_settings.py

```diff
@@ -225,33 +344,84 @@ def __init__(self, **kwargs):
         self.l=kwargs['localization']
 
         self.rpm_override = self.m.spindle_cooldown_rpm_override
-        self.spindle_cooldown_speed.bind(focus=self.on_focus)
+        self.cooldown_speed_slider.bind(value=self.cooldown_speed_updated)
+        self.cooldown_time_slider.bind(value=self.cooldown_time_updated)
+
+        self.spindle_save_widget = widget_maintenance_spindle_save.SpindleSaveWidget(machine=self.m, screen_manager=self.sm, localization=self.l)
+        self.spindle_save_container.add_widget(self.spindle_save_widget)
 
         self.update_strings()
 
-    def on_focus(self, instance, value):
-        if not value:
-            self.rpm_override = True
+    def cooldown_speed_updated(self, instance, value):
+        # Convert to int and display
+        self.rpm_label.text = "%i " % value + self.l.get_str("RPM")
+        self.rpm_override = True
+
+    def cooldown_time_updated(self, instance, value):
+        # Convert to int and display
+        self.seconds_label.text = "%i " % value + self.l.get_str("seconds")
```

Comment:
It's worth checking that all of the value + unit labels look right when translated, just in case the spacing needs adjusting :)

Reply:
Adjusted everything to fit in every language